### PR TITLE
PD-928 Enhance pxdoc-tooling to deploy PR to the staging server

### PR DESCRIPTION
**Pull request by dsupure**
**What this PR does / why we need it**:
PD-029 - Enhance pxdoc-tooling to deploy PR to the staging server
**Which issue(s) this PR fixes** (optional)
https://portworx.atlassian.net/browse/PD-928

**Special notes for your reviewer**:

### CODE REVIEW

File: build/build.sh [hunk 1]
```diff
@@ -81,11 +82,25 @@ export DEPLOYMENT_IMAGE_PREFIX=$(yq e ".$YAML_SECTION_NAME.DEPLOYMENT_IMAGE_PREF
...
 export GCP_PROJECT_ID=production-apps-210001
 export GCP_ZONE=us-west1-b
+
+export GCP_STAGING_CLUSTER_ID=pxdocs-staging
+export GCP_STAGING_PROJECT_ID=portworx-eng
+export GCP_STAGING_ZONE=us-central1-c
+
 # Docker builds cannot use uppercase characters in the image name
 export LOWER_CASE_BRANCH=$(echo -n $TRAVIS_BRANCH | awk '{print tolower($0)}')
+
```
Comment (nathan-px) [inline build/build.sh:96]:
```suggestion
# Set the env vars to the staging values
```

File: deploy/manifests/staging/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+              path: /
+              port: 80
+            initialDelaySeconds: 10
+            timeoutSeconds: 1
+          readinessProbe:
+            httpGet:
+              path: /
+              port: 80
+            initialDelaySeconds: 10
+            timeoutSeconds: 1
```
Comment (Pure-AdamuKaapan) [inline deploy/manifests/staging/deployment.yaml:37]: [nit] newlines at the end of files

File: deploy/manifests/staging/service-template.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: docs-${TRAVIS_PULL_REQUEST}
+  namespace: pxdocs
+spec:
+  type: ClusterIP
```
Comment (Pure-AdamuKaapan) [inline deploy/manifests/staging/service-template.yaml:7]: [nit] My test cluster is using NodePorts, though it prooooobably works just fine with ClusterIPs. I'm fine merging like this and we can change later if it becomes an issue.

**Review by Pure-AdamuKaapan — APPROVED** [review]
LGTM!

**Pull request closed without merging**